fnet/src/vespa/fnet/frt/require_capabilities.cpp

```diff
@@ -5,9 +5,25 @@
 #include <vespa/fnet/connection.h>
 #include <vespa/vespalib/net/connection_auth_context.h>
 
+#include <vespa/log/bufferedlogger.h>
+LOG_SETUP(".fnet.frt.require_capabilities");
+
+using namespace vespalib::net::tls;
+
 bool
 FRT_RequireCapabilities::allow(FRT_RPCRequest& req) const noexcept
 {
     const auto& auth_ctx = req.GetConnection()->auth_context();
-    return auth_ctx.capabilities().contains_all(_required_capabilities);
+    const bool is_authorized = auth_ctx.capabilities().contains_all(_required_capabilities);
+    if (!is_authorized) {
+        auto peer_spec = req.GetConnection()->GetPeerSpec();
+        std::string method_name(req.GetMethodName(), req.GetMethodNameLen());
+        LOGBT(warning, peer_spec, "Permission denied for RPC method '%s'. "
```

Comment:
This may result in severe logging if a client uses non-persistent connections

Reply:
Good point. Will look at adding an API for getting a "reduced" spec once @havardpe is back. This will suffice for now.